### PR TITLE
Add unit tests

### PR DESCRIPTION
**Pull request by baileywickham**

### CODE REVIEW

File: .github/workflows/pythonapp.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ name: Python application
 
 on:
   push:
-    branches: [ master ]
+    branches: [ master, unittest ]
```
Comment (baileywickham) [inline .github/workflows/pythonapp.yml]: I need to remove this on push

Comment (Joshua-Anderson) [reply .github/workflows/pythonapp.yml]: Why not just build every branch?

File: tests/polyxdr_test.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    def test_xp(self):
+        generate_schema_output("xp")
+        self.assertTrue(filecmp.cmp("expected/expected.xp", "output/xp.out.xp", shallow=False))
+
+    def test_libproc(self):
+        generate_schema_output("libproc")
+        self.assertTrue(filecmp.cmp("expected/expected.c", "output/libproc.out.c", shallow=False))
+        self.assertTrue(filecmp.cmp("expected/expected.h", "output/libproc.out.h", shallow=False))
+
+#   def test_python(self):
```
Comment (Joshua-Anderson) [inline tests/polyxdr_test.py]: Remove commented out code

**Review by Joshua-Anderson — APPROVED** [review]
LGTM! Nice and easy way to approach the problem. Just had a few nitpick comments that you can take or leave

**Pull request merged**